### DISCUSSION TITLE
mpy-cross: can I compile a module that is a directory containing files as a whole?

### DISCUSSION THREAD

**Discussion by rkompass** [micropython/micropython#11726] [General]
Can I compile a module that is a _directory containing files_ as a whole? I just found out that I can't.
But this would be a nice feature.
My dream would be a directory (`/lib`) full of `.mpy`s of those important libraries that are not compiled into the firmware, automatically generated in the build process. One `.mpy` for each library, not hierarchies of directories.

**Comment by jimmo** [comment]
Can you clarify what you mean?

* You want to compile a directory of .py files into a _single_ .mpy file.  (This is not possible. `.mpy` and `.py` files have a 1:1 correspondence)

or

* You want a fast way to take a directory of .py files and turn it into a directory of .mpy files? (I'd just use find|xargs)

or

* Something else to do with the freezing process (but freezing already handles directories of .py files)

FYI we're working on adding support for using mpy-cross automatically in mpremote. i.e. you'll be able to automatic .py to .mpy conversion in file copy.

**Reply by rkompass** [reply]
> This is not possible. .mpy and .py files have a 1:1 correspondence

I wished it was possible, so that there could be just one .mpy-file for the library (for an elder man who prefers simplicity :-)), but I see that the 1:1 correspondence is an important point to be preserved.
Perhaps a `.mpl` format for a bundle of .mpy's with or without directory structure?
Thanks for your answer. I think of making a bash script with find|xargs that compiles a set of libraries I have in a list, that are not in `help('modules')` into a /lib hierarchy that I can upload to the flash.

**Reply by robert-hh** [reply]
You could in a bash script copy all .py files into one and run mpy-cross on that one.
